compiler/domain.py

```diff
@@ -1,84 +1,162 @@
 """Domain class for the compiler service."""
 
-from typing import NamedTuple, Optional
+from typing import NamedTuple, Optional, BinaryIO, Dict
 import io
 from datetime import datetime
 from .util import ResponseStream
+from enum import Enum
 
 
-class CompilationStatus(NamedTuple):
-    """Represents the state of a compilation product in the store."""
+class Format(Enum):
+    """Compilation formats supported by this service."""
+
+    PDF = "pdf"
+    DVI = "dvi"
+    PS = "ps"
+
+    @property
+    def ext(self) -> str:
+        """Filename extension for the compilation product."""
+        value: str = self.value
+        return value
+
+    @property
+    def content_type(self) -> str:
+        """The mime-type for this format."""
+        _ctypes: Dict['Format', str] = {
+            Format.PDF: 'application/pdf',
+            Format.DVI: 'application/x-dvi',
+            Format.PS: 'application/postscript'
+        }
+        return _ctypes[self]
+
 
-    # These are intended as fixed class attributes, not slots.
-    PDF = "pdf"   # type: ignore
-    DVI = "dvi"   # type: ignore
-    PS = "ps"   # type: ignore
+class Status(Enum):
+    """Represents the status of a requested compilation."""
 
-    CURRENT = "current"   # type: ignore
-    IN_PROGRESS = "in_progress"   # type: ignore
-    FAILED = "failed"   # type: ignore
+    COMPLETED = "completed"
+    IN_PROGRESS = "in_progress"
+    FAILED = "failed"
+
+
+class Reason(Enum):
+    """Specific reasons for a (usually failure) outcome."""
+
+    AUTHORIZATION = "auth_error"
+    MISSING = "missing_source"
+    SOURCE_TYPE = "invalid_source_type"
+    CORRUPTED = "corrupted_source"
+    STORAGE = "storage"
+    CANCELLED = "cancelled"
+    ERROR = "compilation_errors"
```

Comment:
`COMPILATION` might be more succinct than `ERROR`.

Reply:
Changed in fdf74b2